### DISCUSSION TITLE
how to build?

### DISCUSSION THREAD

**Discussion by bobaoapae** [cefsharp/CefSharp#3679] [Q&A]
Hi guys, i'm trying to make some changes on cefsharp but can't figure out how to compile...

i follwed instructions on WIKI page but don't worke for me.

Need some others pre build config?

I make checkout on master, VS 2019 updated, open solution try to build all (a lot of errors).

Tried build from build.bat (a lot of errors).

Tried build from build.ps1 (a lot of errors).
[build-output.txt](https://github.com/cefsharp/CefSharp/files/6792237/build-output.txt)

Tried build from build.netcore.ps1 (a lot of errors).
[netcore-output.txt](https://github.com/cefsharp/CefSharp/files/6792216/netcore-output.txt)

**Comment by amaitland** [comment]
Your text files don't seem to have any actual errors in them.

You need the native desktop(c++ and visual c++), components, win 10sdk etc.

Visual studio should be able to install the components for you see https://devblogs.microsoft.com/setup/configure-visual-studio-across-your-organization-with-vsconfig/

https://github.com/cefsharp/CefSharp/blob/master/.vsconfig

**Reply by bobaoapae** [reply]
aleardy installed, i open solution with visual studio and installed components, but did work i can't compile, maybe something missing but can't figure out what

![Screenshot_22](https://user-images.githubusercontent.com/33332428/125127827-817b6100-e0d3-11eb-926d-59488cc0b395.png)

**Reply by amaitland** [reply]
How exactly did you install the components? The default target when you open visual studio is ARM64, make sure you select either x64 or Win32.

**Comment by bobaoapae** [comment]
![Screenshot_23](https://user-images.githubusercontent.com/33332428/125128328-47f72580-e0d4-11eb-8e81-6d7a8c7efb7e.png)

**Reply by amaitland** [reply]
Change the build output to normal and find the first error in the build output window. 

It's likely you are missing a required component.

**Reply by amaitland** [reply]
Make sure the Nuget restore was successful.

**Reply by bobaoapae** [reply]
Nuget restore successfull

![Screenshot_24](https://user-images.githubusercontent.com/33332428/125141192-b0510180-e0ea-11eb-9c16-203aa4fa44e3.png)

**Comment by bobaoapae** [comment]
never mind, clone repo again and now it's compiling.

Thanks for your attention
